py/dml/messages.py

```diff
@@ -820,6 +820,13 @@ def log(self):
         if self.othersite:
             self.print_site_message(self.othersite, "conflicting declaration")
 
+class EAUNKDIM(DMLError):
+    """
+    The size of an array dimension of an object array must be defined at least
+    once across all declarations of that object array.
+    """
+    fmt = "the size of dimension %d is never defined"
```

Comment:
maybe clearer to mention the index variable name? that's ambiguous only in the not so common case where you have the same variable name in multiple dimensions, but even then the site clears the ambiguity. Dimension number susceptible to off-by-one: is the first dim 0 or 1?

Reply:
How about both?
"the size of dimension %d (with index variable '%s') is never defined"

Reply:
sure